arangod/Cluster/ClusterInfo.cpp

```diff
@@ -4831,22 +4822,21 @@ std::unordered_map<ServerID, std::string> ClusterInfo::getServers() {
     loadServers();
   }
   READ_LOCKER(readLocker, _serversProt.lock);
-  std::unordered_map<ServerID, std::string> serv = _servers;
-  return serv;
+  return _servers;
```

Comment:
Is this guaranteed to be copied with lock held? I remember we have once similar problems with `VPackObjectBuilder` that accessed in their destructor a move away builder. This was only on windows, but worked with gcc + clang. i.e. on gcc clang the destructors ran before the return value was moved.

Reply:
If the copy is _not_ made with the lock being correctly held, then basically all C++ RAII patterns would be broken, too.

But I remember the problem we had once with the VPackObjectBuilder in some build modes on Windows... I think the problem there was something like that:
```
VPackBuilder func() {
  VPackBuilder result;
  VPackObjectBuilder obj(&result);
  result.add(...);
  return result;
}
```
I think this is a bit different, because the object builder referred to the (moved-away or optimized-away) variable.
In contrast, an RW-lock or a mutex will not lose any of its functionality if we move/optimize away the returned object of a different kind.
So I think this **must** be safe. Otherwise, we would have a lot of problems, in others places in ClusterInfo.cpp (e.g. https://github.com/arangodb/arangodb/blob/3.6/arangod/Cluster/ClusterInfo.cpp#L4380) and everywhere else.